vm/src/obj/objset.rs

```diff
@@ -3,32 +3,67 @@
  */
 
 use super::super::pyobject::{
-    IdProtocol, PyContext, PyFuncArgs, PyObject, PyObjectPayload, PyObjectRef, PyResult,
-    TypeProtocol,
+    PyContext, PyFuncArgs, PyObject, PyObjectPayload, PyObjectRef, PyResult, TypeProtocol,
 };
 use super::super::vm::VirtualMachine;
 use super::objbool;
 use super::objiter;
 use super::objstr;
 use super::objtype;
+use num_bigint::BigInt;
 use num_bigint::ToBigInt;
 use std::collections::HashMap;
 
-pub fn get_elements(obj: &PyObjectRef) -> HashMap<usize, PyObjectRef> {
+pub fn get_elements(obj: &PyObjectRef) -> HashMap<BigInt, PyObjectRef> {
     if let PyObjectPayload::Set { elements } = &obj.borrow().payload {
         elements.clone()
     } else {
         panic!("Cannot extract set elements from non-set");
     }
 }
 
-pub fn sequence_to_hashmap(iterable: &Vec<PyObjectRef>) -> HashMap<usize, PyObjectRef> {
-    let mut elements = HashMap::new();
-    for item in iterable {
-        let key = item.get_id();
-        elements.insert(key, item.clone());
+fn perform_action_with_hash(
+    vm: &mut VirtualMachine,
+    elements: &mut HashMap<BigInt, PyObjectRef>,
+    item: &PyObjectRef,
+    f: &Fn(
```

Comment:
Is there a way to rewrite this code such that this function pass can be avoided? In my opinion creating a local function and passing that function is very powerful, but might be harder to understand what is going on.